libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -272,6 +272,11 @@ AP_GPS::detect_instance(uint8_t instance)
             hal.console->print_P(PSTR(" SBP "));
             new_gps = new AP_GPS_SBP(*this, state[instance], _port[instance]);
         }
+        else if ((_type[instance] == GPS_TYPE_AUTO || _type[instance] == GPS_TYPE_SBF) &&
```

Comment:
I think this driver should be like the PX4 GPS driver where you must set GPS_TYPE to the right value (ie. not run detect if it is GPS_TYPE_AUTO)